tests/virt_autotest/virsh_external_snapshot.pm

```diff
@@ -51,6 +51,10 @@ sub run_test {
     my @vm_hostnames_inactive_array = split(/\n+/, $vm_hostnames_inactive);
 
     foreach my $guest (keys %xen::guests) {
+        #RAW do not support Snapshot test
+        my $guest_disk_type = script_output("virsh dumpxml $guest | grep 'driver name' | sed \"s/'/ /g\"| awk \'{ print \$5 }\'");
```

Comment:
the line in <vm.xml> must keep the current format strictly then you can get the correct disk type. It is a little unsafe. such as we put more spaces, or reverse the attributes sequence etc. you'd better just get the 'type' more precisely in the line. 
<driver name='qemu' type='qcow2'/>

Reply:
yes, good point. I had spent most time to get the 'type' more precisely by Regular Expression way to deal with the more spaces and so on as your mentioned , such as used with the sed and awk together.

Reply:
such as this one works even the attribute sequence may change or more attributes are added in the output:
virsh dumpxml $guest | grep '<driver ' | grep -o "type='.*'" | cut -d "'" -f2   //need character escape in script_output().

Reply:
Great, thank you so much for your great. 
I will update code here as your mentioned.

Reply:
had updated code already